src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -581,21 +586,36 @@ private async Task WhenContextCreated(CancellationToken cancellationToken)
     /// </summary>
     private sealed class EvaluationDataAdapter : EvaluationData
     {
-        private readonly IImmutableDictionary<string, string> _properties;
+        private ProjectInstance? _projectInstance;
 
-        public EvaluationDataAdapter(IImmutableDictionary<string, string> properties)
+        public EvaluationDataAdapter(ProjectInstance projectInstance)
         {
-            _properties = properties;
+            _projectInstance = projectInstance;
         }
 
         public override string GetPropertyValue(string name)
         {
-            _properties.TryGetValue(name, out string? value);
+            if (_projectInstance is null)
+            {
+                throw new ObjectDisposedException(nameof(EvaluationDataAdapter));
+            }
+
+            string? value = _projectInstance.GetProperty(name)?.EvaluatedValue;
 
             // Return the empty string rather than null.
             value ??= "";
 
             return value;
```

Comment:
nit: Reassigning the local value provides no purpose. Potentially remove the comment too, since it says nothing other than what the code itself says.
```suggestion
            // Return the empty string rather than null.
            return value ?? "";
```
Or even more terse would be just `return _projectInstance.GetProperty(name)?.EvaluatedValue ?? "";` since the variable isn't used for anything other than returning.